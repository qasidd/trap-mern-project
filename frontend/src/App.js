import './resources/App.css';

import Discussionboard from './components/pages/Discussionboard';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import Classifications from './components/Classifications/Classifications';
import ContactForm from './components/Contactsus/contactform';
import ListingsGallery from './components/Galleries/ListingsGallery';
import NewReleasesGallery from './components/Galleries/NewReleasesGallery';
import Home from './components/Home/Home';
import OpeningTime from './components/Opening-Times/CinemaTime';
import './resources/App.css';
import Nav from "./components/Routing/Nav";
import About from './components/About/About';
import React from "react";
import { ImageBackground, StyleSheet, Text, View } from "react";
import GettingThere from './components/Getting-There/GettingThere';
import PlacesToGo from './components/PlacesToGo/PlacesToGo';
import NewBooking from './components/Booking/NewBooking';
import Search from './components/SearchBar/Search';
import Searchable from './components/SearchBar/Searchable';
import PaymentSuccess from './components/Booking/PaymentSuccess';
import PaymentCancelled from './components/Booking/PaymentCancelled';
import Screenparent from './components/Screens/screenparent';

function App() {
  return (
    <div className="App">
      <Router>
        <Nav />
        <Switch>
          <Route path="/" exact>
            <Home />
          </Route>
          <Route path="/about">
            <About />
          </Route>
          <Route path="/listings-gallery">
            <ListingsGallery/>
          </Route>
          <Route path="/new-releases">
            <NewReleasesGallery/>
          </Route>
          <Route path="/opening-times">
            <OpeningTime/>
          </Route>
          <Route path="/classification">
            <Classifications/>
          </Route>
          <Route path="/screens">
            <Screenparent/>
          </Route>
          <Route path="/new-booking" exact>
            <NewBooking />
          </Route>
          <Route path="/location" exact>
            <GettingThere/>
          </Route>
          <Route path="/placestogo" exact>
            <PlacesToGo/>
          </Route>
          <Route path="/contactus" exact>
            <ContactForm/>
          </Route>
          <Route path="/discussion" exact>
            <Discussionboard/>
          </Route>
          <Route path="/search">
            <Searchable />
          </Route>
          <Route path="/new-booking/success" component={PaymentSuccess} />
          <Route path="/new-booking/cancelled" component={PaymentCancelled} />
        </Switch>
      </Router>
      {/* <Searchable/> */}
     
    </div>
  );
}


// Random text so I can commit 
export default App;
