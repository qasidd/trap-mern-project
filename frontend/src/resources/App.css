:root {
    --main-bg-colour: #9B9898;
    --main-accent-colour: #FF6033
}

body {
    background-color: var(--main-bg-colour);
}

.mainContent {
    padding-top: 1em;
}

.mainContent h1 {
    color: var(--main-accent-colour);
}

a {
    color: var(--main-accent-colour) !important;
}

.mainContent .nav-item a {
    color: var(--main-accent-colour);
}

.galleryPoster {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-bottom: 1rem;
    padding: 0px 2rem;
    text-align: center;
    border-radius: 20px;
    transition: transform .2s;
    color: white;
}

.galleryPoster span {
    visibility: hidden;
    font-size: 22px;
    font-weight: 100;
    text-shadow: 0px 1px rgba(0,0,0);
}

.galleryPoster:hover {
    transform: scale(1.02);
    border: 5px solid var(--main-accent-colour);
    box-shadow: inset 0px 0px 300px 10px rgba(0,0,0,0.8);
}

.galleryPoster:hover span {
    visibility: visible;
}

.filmModal .modal-body {
    background-color: var(--main-accent-colour);
    color: white;
}

.filmModal .modal-header {
    color: white;
    background-color: #c44824;
    border-bottom: 0;
}

.filmModal .modal-body img {
    position: absolute;
    z-index: 1;
    margin-left: 66%;
    margin-top: -150px;
    border-radius: 15px;
    max-height: 200px;
}

