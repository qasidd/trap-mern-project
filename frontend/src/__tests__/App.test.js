import App from '../App';
import '@testing-library/jest-dom';
import {create} from 'react-test-renderer';
import { expect, it } from '@jest/globals';



//Snapshot test: 
describe(`Snapshot test of APP`, () => {
  it(`it should match the snapshot everytime it's rendered`, () => {
    const instance = create(<App/>).toJSON();
    expect (instance).toMatchSnapshot();

  })

});

