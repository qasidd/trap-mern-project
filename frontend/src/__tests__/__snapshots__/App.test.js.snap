// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of APP it should match the snapshot everytime it's rendered 1`] = `
<div
  className="App"
>
  <nav
    class="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      class="navbar-brand"
    >
      QA Cinemas
    </a>
    <button
      aria-controls="navbarNavAltMarkup"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-target="#navbarNavAltMarkup"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        class="navbar-nav"
      >
        <a
          className="btn btn-outline-info"
          href="/"
          onClick={[Function]}
        >
          Home 
          <span
            class="sr-only"
          >
            (current)
          </span>
        </a>
        <a
          className="btn btn-outline-info"
          href="/listings-gallery"
          onClick={[Function]}
        >
          Gallery
        </a>
        <a
          className="btn btn-outline-info"
          href="/new-releases"
          onClick={[Function]}
        >
          New Releases
        </a>
        <a
          className="btn btn-outline-info"
          href="/opening-times"
          onClick={[Function]}
        >
          Opening Times
        </a>
        <a
          className="btn btn-outline-info"
          href="/classification"
          onClick={[Function]}
        >
          Classifications
        </a>
        <a
          className="btn btn-outline-info"
          href="/screens"
          onClick={[Function]}
        >
          Screens
        </a>
        <a
          className="btn btn-outline-info"
          href="/new-booking"
          onClick={[Function]}
        >
          Bookings
        </a>
        <a
          className="btn btn-outline-info"
          href="/location"
          onClick={[Function]}
        >
          Getting Here
        </a>
        <a
          className="btn btn-outline-info"
          href="/placestogo"
          onClick={[Function]}
        >
          Places to Go
        </a>
        <a
          className="btn btn-outline-info"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="btn btn-outline-info"
          href="/contactus"
          onClick={[Function]}
        >
          Contact Us
        </a>
        <a
          className="btn btn-outline-info"
          href="/discussion"
          onClick={[Function]}
        >
          Discussion Board
        </a>
      </div>
    </div>
  </nav>
  <h1>
     Welcome to QA Cinemas! 
  </h1>
  <br />
  <img
    alt="this is a logo"
    src="logo.jpg"
  />
  <h2>
     Redefining the cinema experience 
  </h2>
  <br />
  <p>
     This is the homepage for QA cinemas. This page will provide information about movies, listings and upcoming releases. 
  </p>
  <h3>
     Page Under Construction 
  </h3>
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
</div>
`;
